crates/database/src/jit_orders.rs

```diff
@@ -0,0 +1,262 @@
+    jit_orders: &[JitOrder],
+) -> Result<(), sqlx::Error> {
+    if jit_orders.is_empty() {
+        return Ok(());
+    }
+
+    let mut query_builder = QueryBuilder::new(
+        r#"
+        INSERT INTO jit_orders (
+            block_number,
+            log_index,
+            uid,
+            owner,
+            creation_timestamp,
+            sell_token,
+            buy_token,
+            sell_amount,
+            buy_amount,
+            valid_to,
+            app_data,
+            fee_amount,
+            kind,
+            partially_fillable,
+            signature,
+            receiver,
+            signing_scheme,
+            sell_token_balance,
+            buy_token_balance
+        ) 
+        "#,
+    );
+
+    query_builder.push_values(jit_orders.iter(), |mut builder, jit_order| {
+        builder
+            .push_bind(jit_order.block_number)
+            .push_bind(jit_order.log_index)
+            .push_bind(jit_order.uid)
+            .push_bind(jit_order.owner)
+            .push_bind(jit_order.creation_timestamp)
+            .push_bind(jit_order.sell_token)
+            .push_bind(jit_order.buy_token)
+            .push_bind(jit_order.sell_amount.clone())
+            .push_bind(jit_order.buy_amount.clone())
+            .push_bind(jit_order.valid_to)
+            .push_bind(jit_order.app_data)
+            .push_bind(jit_order.fee_amount.clone())
+            .push_bind(jit_order.kind)
+            .push_bind(jit_order.partially_fillable)
+            .push_bind(jit_order.signature.clone())
+            .push_bind(jit_order.receiver)
+            .push_bind(jit_order.signing_scheme)
+            .push_bind(jit_order.sell_token_balance)
+            .push_bind(jit_order.buy_token_balance);
+    });
+
+    query_builder.push(
+        r#"
+        ON CONFLICT (block_number, log_index) DO UPDATE 
+        SET uid = EXCLUDED.uid,
+            owner = EXCLUDED.owner,
+            creation_timestamp = EXCLUDED.creation_timestamp,
+            sell_token = EXCLUDED.sell_token,
+            buy_token = EXCLUDED.buy_token,
+            sell_amount = EXCLUDED.sell_amount,
+            buy_amount = EXCLUDED.buy_amount,
+            valid_to = EXCLUDED.valid_to,
+            app_data = EXCLUDED.app_data,
+            fee_amount = EXCLUDED.fee_amount,
+            kind = EXCLUDED.kind,
+            partially_fillable = EXCLUDED.partially_fillable,
+            signature = EXCLUDED.signature,
+            receiver = EXCLUDED.receiver,
+            signing_scheme = EXCLUDED.signing_scheme,
+            sell_token_balance = EXCLUDED.sell_token_balance,
+            buy_token_balance = EXCLUDED.buy_token_balance;
+        "#,
+    );
+
+    let query = query_builder.build();
+    query.execute(ex).await?;
+
+    Ok(())
+}
+
+/// Jit order combined with trades table and order_execution table, suitable for
+/// API responses.
+#[derive(Debug, Clone, Default, PartialEq, sqlx::FromRow)]
+struct FullJitOrder {
```

Comment:
nit
```suggestion
struct JitOrderWithExecutions {
```

Full is a bit vague